bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoAbstractHandler.java

```diff
@@ -335,9 +336,11 @@ protected boolean initializeData() {
 
     /**
      * Prepares the ExpiringCache for network data
+     *
+     * @return
      */
-    protected void initalizeNetworkCache() {
-        network = new ExpiringCache<String>(CACHE_EXPIRY_NETWORK, () -> {
+    protected ExpiringCache<String> initalizeNetworkCache() {
+        final ExpiringCache<String> network = new ExpiringCache<String>(CACHE_EXPIRY_NETWORK, () -> {
```

Comment:
Why can't this created in the constructor? That way you can make the `network` field non-null and final.

Reply:
yes, that's a good way to solve it. Done.